1.train_models/log_reg_plates_3_3p_5_cp_norm_data/nbconverted/plate_3_3p_5_train_log_reg_scv.py

```diff
@@ -0,0 +1,365 @@
+
+    Returns
+    -------
+    _df: Pandas Dataframe
+        Cleaned plated data with nans and HET cells removed.
+    """
+
+    _df.dropna(inplace=True)
+    _df = _df.loc[_df[gene_column] != "HET"]
+    return _df
+
+
+# ## Split and process plates
+# We aim to maximize the the number of cells in the train-validation set per plate.
+# We achieve this by select the specific holdout wells that maximize the minority class in the train-validation set.
+# In other words, we choose the combination of wells for train-validation that, together, include the highest number of cells in the genotype category that has the fewest number of cells.
+# By side-effect, this process also minimizes the number of cells dropped from training in our downsampling procedure to balance datasets for class size prior to model training.
+
+# In[6]:
+
+
+plate5df = process_plates(plate5df)
+rest5df, test5df = split_plates(plate5df, 4)
+rest5df, test5df = down_sample_by_genotype(rest5df), down_sample_by_genotype(test5df)
+num_test = test5df.shape[0]
+print(f"Fraction of test cells plate 5 = {num_test / (num_test + rest5df.shape[0])}\n")
+
+plate3df = process_plates(plate3df)
+rest3df, test3df = split_plates(plate3df, 7)
+rest3df, test3df = down_sample_by_genotype(rest3df), down_sample_by_genotype(test3df)
+num_test = test3df.shape[0]
+print(f"Fraction of test cells plate 3 = {num_test / (num_test + rest3df.shape[0])}\n")
+
+plate3pdf["Metadata_Plate"] = "Plate_3p"
+plate3pdf = process_plates(plate3pdf)
+rest3pdf, test3pdf = split_plates(plate3pdf, 5)
+rest3pdf, test3pdf = down_sample_by_genotype(rest3pdf), down_sample_by_genotype(test3pdf)
+num_test = test3pdf.shape[0]
+print(f"Fraction of test cells plate 3 prime = {num_test / (num_test + rest3pdf.shape[0])}\n")
+
+
+# ## Harmonize data across plates to each data split
+
+# In[7]:
+
+
+# Columns common to all plates
+plate_cols = list(set(plate5df.columns) & set(plate3df.columns) & set(plate3pdf.columns))
+
+restdf = pd.concat([rest5df[plate_cols], rest3df[plate_cols], rest3pdf[plate_cols]], ignore_index=True)
+
+testdf = pd.concat([test5df[plate_cols], test3df[plate_cols], test3pdf[plate_cols]], ignore_index=True)
+
+
+# ## Encode genotypes and extract feature data
+
+# In[8]:
+
+
+le = LabelEncoder()
+
+y = le.fit_transform(restdf["Metadata_genotype"])
+X = restdf.drop(columns=meta_cols)
+
+
+# # Train Models
+
+# ## Specify parameters for training
+
+# In[9]:
+
+
+logreg_params = {
+    "max_iter": 250,
+    "random_state": 0,
+    "n_jobs": -1,
+    "penalty": "l2",
+}
+
+# Random sampling range of hyperparameter
+param_ranges = {
+    "C": (0, 200)
+}
+
+# Number of iteration to optimize hyperparameters
+rand_iter = 500
+
+# Best accuracy
+best_acc = 0
+
+# Initial accuracy
+acc = 0
+
+# Number of folds
+n_splits = 8
+
+
+# Generate hyperparameter samples
+random_params = {
+    i:
+    {key: random.uniform(*param_ranges[key]) for key in param_ranges}
```

Comment:
Thanks for the suggestion, this can be very helpful. Here, I just wanted to make code that could account for multiple hyperparameters for use in future analysis, but it may be easier to read that way.